main.go

```diff
@@ -371,19 +372,22 @@ func main() {
 		os.Exit(1)
 	}
 
-	collectorReconciler := controllers.NewReconciler(controllers.Params{
-		Client:   mgr.GetClient(),
-		Log:      ctrl.Log.WithName("controllers").WithName("OpenTelemetryCollector"),
-		Scheme:   mgr.GetScheme(),
-		Config:   cfg,
-		Recorder: mgr.GetEventRecorderFor("opentelemetry-operator"),
-		Reviewer: reviewer,
-		Version:  v,
-	})
+	var collectorReconciler *controllers.OpenTelemetryCollectorReconciler
+	if cfg.CollectorAvailability() == collector.Available {
```

Comment:
This changes current behavior in a way I don't think is desirable. Crashing isn't good, so this is a useful detection mechanism, but the operator should still exit with an error if it doesn't find the CRDs it expects. For the vast majority of users, this will be a misconfiguration, and it will be confusing for them if the operator just keeps running without doing the work.

I'm fine with adding a flag later to change this behaviour, but we shouldn't change it by default.

Reply:
ok, I see what you're saying. I can make this consistent.